File: algorithms/2149.rearrange-array-elements-by-sign/solution.rs
impl Solution {
    pub fn rearrange_array(nums: Vec<i32>) -> Vec<i32> {
        let mid = nums.len() / 2;
        let (mut pos, mut neg) = (vec![], vec![]);
        for n in nums {
            if n > 0 {
                pos.push(n);
            } else {
                neg.push(n);
            }
        }

        let mut res = vec![];
        for i in 0..mid {
            res.push(pos[i]);
            res.push(neg[i]);
        }

        res
    }
}

File: algorithms/2149.rearrange-array-elements-by-sign/solution.go
func rearrangeArray(nums []int) []int {
	pos, neg := []int{}, []int{}

	for _, n := range nums {
		if n > 0 {
			pos = append(pos, n)
		} else {
			neg = append(neg, n)
		}
	}

	res := []int{}
	for i := 0; i < len(nums)/2; i++ {
		res = append(res, pos[i], neg[i])
	}
	return res
}
